packages/store/src/IStore.sol

```diff
@@ -6,23 +6,24 @@ import { PackedCounter } from "./PackedCounter.sol";
 import { FieldLayout } from "./FieldLayout.sol";
 import { Schema } from "./Schema.sol";
 import { IStoreHook } from "./IStoreHook.sol";
+import { ResourceId } from "./ResourceId.sol";
 
 interface IStoreRead {
   event HelloStore(bytes32 indexed storeVersion);
 
   function storeVersion() external view returns (bytes32);
 
-  function getFieldLayout(bytes32 tableId) external view returns (FieldLayout fieldLayout);
+  function getFieldLayout(ResourceId tableId) external view returns (FieldLayout fieldLayout);
```

Comment:
Did you decide against calling this type TableId and having specific types for each resource type?

Reply:
Unfortunately that would make it much harder to share the `Instance` libraries between the different types (and since all resource ids are structured the same way i think it makes sense to call the type the more general `ResourceId`

Reply:
ahhh I see now that `ResourceId` is generic name+type and `WorldResourceId` is generic namespace+name+type

Reply:
Yeah! They share the same user type `ResourceId`, but by doing `using WorldResourceIdInstance for ResourceId` we get `getNamespace`, `getName`, etc (while `getType` is on `ResourceIdInstance`)

Reply:
woah TIL you can "extend" like that